eZ/Publish/Core/Search/Common/Slot/PublishContentTypeDraft.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Search\Common\Slot;
+
+use eZ\Publish\Core\Search\Common\Slot;
+use eZ\Publish\Core\SignalSlot\Signal;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+
+class PublishContentTypeDraft extends Slot
+{
+    /**
+     * Receive the given $signal and react on it.
+     *
+     * @param Signal $signal
+     */
+    public function receive(Signal $signal)
+    {
+        if (!$signal instanceof Signal\ContentTypeService\PublishContentTypeDraftSignal) {
+            return;
+        }
+
+        $contentTypeService = $this->repository->getContentTypeService();
+        // loadContentType, not loadContentTypeDraft, because draft has been already published
+        $contentType = $contentTypeService->loadContentType($signal->contentTypeDraftId);
+        $criterion = new Criterion\ContentTypeId($contentType->id);
+        $query = new Query(['filter' => $criterion]);
+        $results = $this->searchHandler->findContent($query);
+
+        foreach ($results->searchHits as $searchHit) {
+            $contentInfo = $searchHit->valueObject;
+            $this->searchHandler->indexContent($this->persistenceHandler->contentHandler()->load($contentInfo->id, $contentInfo->currentVersionNo));
```

Comment:
In this new `PublishContentTypeDraft` Slot, there are two issues.

1. Performance - there could be a lot of Content Objects to index after modifying Content Type.
2. Elasticsearch test [adding field definition](https://github.com/ezsystems/ezpublish-kernel/blob/6.4/eZ/Publish/API/Repository/Tests/FieldType/BaseIntegrationTest.php#L960) for `\eZ\Publish\API\Repository\Tests\FieldType\MapLocationIntegrationTest` [fails](https://travis-ci.org/ezsystems/ezpublish-kernel/jobs/160979931#L773) due to this operation. 

My local `elasticsearch.log` shows
```
[action.index             ] [Arsenic] [ezpublish][4], node[TR8oBHDmTPygSWT9Hne3Mw], [P], s[STARTED]: Failed to execute [index {[ezpublish][content][60], source[{"id":"60","type_id":"36","version_id":"1","status_id":"1","name_s":"Test object Bielefeld","creator_id":"14","owner_id":"14","owner_user_group_mid":["4","12","14"],"section_id":"1","section_identifier_id":"standard","section_name_s":"Standard","remote_id_id":"abcdef0123456789abcdef0123456789","modified_dt":"2016-09-19T09:45:24Z","published_dt":"2016-09-19T09:45:24Z","language_code_ms":"eng-US","always_available_b":true,"locations_doc":[],"group_mid":"1","fields_doc":{"meta_language_code_s":"eng-US","meta_is_main_translation_b":true,"meta_is_always_available_b":true,"test-ezgmaplocation_name_value_s":"Test object","test-ezgmaplocation_name_fulltext_fulltext":"Test object","test-ezgmaplocation_name_fulltext_meta_all_eng_US_fulltext":"Test object","test-ezgmaplocation_data_value_address_s":"","test-ezgmaplocation_data_value_location_gl":{"lat":null,"lon":null},"test-ezgmaplocation_data_fulltext_fulltext":[],"test-ezgmaplocation_data_fulltext_meta_all_eng_US_fulltext":[]},"object_state_mid":"1"}]}]
org.elasticsearch.index.mapper.MapperParsingException: failed to parse
```
For Solr this test passes. I'm not sure what's wrong there but I checked that this is due to `indexContent` called here.
ping @andrerom @pspanja